lab_3_generate_by_ngrams/main.py

```diff
@@ -270,6 +376,22 @@ def run(self, seq_len: int, prompt: str) -> Optional[str]:
         In case of corrupt input arguments or methods used return None,
         None is returned
         """
+        if not isinstance(seq_len, int) or not isinstance(prompt, str) or not prompt:
+            return None
+        encoded_text = self._text_processor.encode(prompt)
+        n_gram_size = self._model.get_n_gram_size()
+        if not encoded_text or not n_gram_size:
+            return None
+        while seq_len > 0:
```

Comment:
why use while instead of for when you know exactly how many steps there will be?